models/lwr/pincell/make_openmc_model.py

```diff
@@ -0,0 +1,149 @@
+t_zr_clad = 0.057
+
+## The thickness of the axial reflector above the lattice.
+reflector_t = 21.42
+
+###############################################################################
+# Create materials for the problem
+
+## UO2 for the fuel.
+uo2_comp = 1.0e24 * np.array([8.65e-4, 2.225e-2, 4.622e-2])
+uo2_frac = uo2_comp / np.sum(uo2_comp)
+uo2 = openmc.Material(name = 'UO2 Fuel', temperature = 293.15)
+uo2.add_nuclide('U235', uo2_frac[0], percent_type = 'ao')
+uo2.add_nuclide('U238', uo2_frac[1], percent_type = 'ao')
+uo2.add_element('O', uo2_frac[2], percent_type = 'ao')
+uo2.set_density('atom/cm3', np.sum(uo2_comp))
+
+## Pure Zirconium for the cladding.
+zr = openmc.Material(name = 'Zr Cladding', temperature = 293.15)
+zr.add_element('Zr', 1.0, percent_type = 'ao')
+zr.set_density('atom/cm3', 1.0e24 * 4.30e-2)
+
+## Borated water.
+h2o_comp = 1.0e24 * np.array([3.35e-2, 2.78e-5])
+h2o_frac = h2o_comp / np.sum(h2o_comp)
+h2o = openmc.Material(name = 'H2O Moderator', temperature = 293.15)
+h2o.add_element('H', 2.0 * h2o_frac[0], percent_type = 'ao')
+h2o.add_element('O', h2o_frac[0], percent_type = 'ao')
+h2o.add_element('B', h2o_frac[1], percent_type = 'ao')
+h2o.set_density('atom/cm3', np.sum(h2o_comp))
+h2o.add_s_alpha_beta('c_H_in_H2O')
+
+# Collect the materials together and export to XML
+materials = openmc.Materials([uo2, zr, h2o])
+materials.export_to_xml()
+
+###############################################################################
+# Define problem geometry
+
+# Create cylindrical surfaces
+fuel_or = openmc.ZCylinder(r = r_fuel, name = 'Fuel Outer Radius')
+clad_ir = openmc.ZCylinder(r = r_fuel + t_f_c_gap, name = 'Clad Inner Radius')
+clad_or = openmc.ZCylinder(r = r_fuel + t_f_c_gap + t_zr_clad, name = 'Clad Outer Radius')
+
+# Create a region represented as the inside of a rectangular prism
+pitch = 1.26
+box = openmc.model.RectangularPrism(pitch, pitch, boundary_type='reflective')
+
+# Create cells, mapping materials to regions - split up the axial height
+planes = np.linspace(0.0, height, N + 1)
+plane_surfaces = []
+for i in range(N + 1):
+  plane_surfaces.append(openmc.ZPlane(z0=planes[i]))
+
+# set the boundary condition on the topmost and bottommost planes to vacuum
+plane_surfaces[0].boundary_type = 'reflective'
+
+fuel_cells = []
+clad_cells = []
+gap_cells = []
+water_cells = []
+all_cells = []
+for i in range(N):
+  layer = +plane_surfaces[i] & -plane_surfaces[i + 1]
+  fuel_cells.append(openmc.Cell(fill=uo2, region=-fuel_or & layer, name='Fuel{:n}'.format(i)))
+  gap_cells.append(openmc.Cell(fill=None, region=+fuel_or & -clad_ir & layer, name='Gap{:n}'.format(i)))
+  clad_cells.append(openmc.Cell(fill=zr, region=+clad_ir & -clad_or & layer, name='Clad{:n}'.format(i)))
+  water_cells.append(openmc.Cell(fill=h2o, region=+clad_or & layer & -box, name='Water{:n}'.format(i)))
+  all_cells.append(fuel_cells[i])
+  all_cells.append(gap_cells[i])
+  all_cells.append(clad_cells[i])
+  all_cells.append(water_cells[i])
```

Comment:
Is there a reason to append these in this order versus appending the full list of `fuel_`, `gap_`, `clad_` and `water_cells` after the loop?